### ISSUE TITLE
R usage seems to make unusual assumptions

### ISSUE THREAD

**Issue by smarr** [softdevteam/warmup_stats#95]
You got the following checks in three of your scripts:

```python
if ('R_LIBS_USER' not in os.environ or 'rlibs' not in os.environ['R_LIBS_USER']):
```

First problem for me is that I do not have `R_LIBS_USER` set.
Yes, the instructions tell me to set it, though, I'd rather prefer to simply install the R libs and then not to have any other things floating around.

The next issue is that even if I set the environment variable, it's ignored, because it doesn't contain the `rlibs` string.

My setup looks like this:
```
> .libPaths()
[1] "/Users/smarr/Library/R/3.5/library"
[2] "/opt/local/Library/Frameworks/R.framework/Versions/3.5/Resources/library"
```

My local solution is to simply disable the code that checks and sets `R_LIBS_USER`, and let R do its thing.
But I suppose you want to use the R libs local to this project. Perhaps you can check against that instead of checking against `R_LIBS_USER`?

**Comment by ltratt** [comment]
It looks like the install instructions are out of date as a result of https://github.com/softdevteam/warmup_stats/pull/70 which, ironically, was intended to make things easier to install. Oops!

I *think* the right thing to do here is to simply run `./build.sh` (which won't build R! but it will build the libraries we need locally) and then things should "just work (TM)".

Does that work for you? If so, I'll update the instructions and probably also change:

```if ('R_LIBS_USER' not in os.environ or 'rlibs' not in os.environ['R_LIBS_USER']):
    os.environ['R_LIBS_USER'] = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
                                             'work', 'rlibs')
```

to something like:

```python
our_rlibs = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'rlibs')
if ('R_LIBS_USER' in os.environ):
    os.environ['R_LIBS_USER'] = "%s:%s" % (os.environ['R_LIBS_USER'], our_rlibs)
else:
    os.environ['R_LIBS_USER'] = our_rlibs
```

so that people with more elaborate R installations don't have to do anything clever either.

**Comment by smarr** [comment]
I would prefer a solution that makes sure that the initial assumption, i.e., that `build.sh` was used, is checked first.
Otherwise, I'd leave R alone.
So, guarding everything with a `if file.exists('work/rlibs')` might be a solution.

The `build.sh` is so full of assumptions that I'd rather install the R libraries and rpy2 manually.

**Comment by smarr** [comment]
Perhaps to give more details about my environment: It's a macOS with macports, and python 3 as default python. So, running `pip` doesn't work directly either.

**Comment by ltratt** [comment]
As things stand, you absolutely have to have Python 2.7 installed (with the `python2.7` binary available) for warmup_stats -- I doubt we'll ever port it to Python 3 TBH.

Note that `build.sh` only modifies things *within* `warmup_stats/work` (significantly, it doesn't install anything globally!), so it should be a very safe script: I don't think there's a good reason not to run it. Put another way, if you want to install things yourself, that should be doable with my suggested fix, because you can then set `R_LIBS_USER` to whatever you want.

**Comment by smarr** [comment]
I do have python 2.7, but python 3 is set as *default*, which means, `pip` is used to install stuff into the python 3 tree. If I would want to use the script, I'd have to change `pip` to `pip-2.7`.

And at least from my perspective, not checking the basic assumption that your local `rlibs` folder exists seems wrong. You force changes onto the system that make it break, which can't be the desired outcome.

**Comment by smarr** [comment]
Also, just as a note, I got a local work-around, so, I don't want you to change your stuff unnecessarily, but I thought it might be worth reporting it.

**Comment by ltratt** [comment]
Hmm, the `pip` thing is a good point. Let me look into what we can do there (if anything...). I also agree that checking that `work/rlibs` exists is a good idea.

**Comment by ltratt** [comment]
And also, thank you for reporting this!

**Comment by ltratt** [comment]
Fixed this to the extent we can in #96.
